portfolio/src/main/webapp/script.js

```diff
@@ -38,3 +38,57 @@ function myFunction() {
     x.className = "nav";
   }
 }
+
+function getHello() {
+  console.log('Fetching a random message.');
+  
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+*/
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+	
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addToDom);
+}
+
+/** Adds a random message to the DOM. */
+function addToDom(message) {
+  console.log('Adding quote to dom: ' + message);
+  const messageContainer = document.getElementById('message-container');
+  messageContainer.innerText = message;
+}
+
+// async version of the get HellowMessage
```

Comment:
Use one commenting style for method documentation. Standard forward method comments would be best so like the comment on `handleResponse`. There's (obviously) also a [Google JavaScript Style](https://google.github.io/styleguide/jsguide.html) for JavaScript.

Reply:
Thanks for pointing it out. This is something I do not pay much attention to.